### PR TITLE
feat(analytics): setting up baseline glean with a dummy event

### PR DESCRIPTION
**Pull request by bassrock**
## Goal

Play with how glean can be added.

Follows the docs here: https://mozilla.github.io/glean/book/user/adding-glean-to-your-project/javascript.html

## Todo
- [ ] Where do we initialize Glean sdk within Vue
    - [ ] Do we use the Node or Web version of Glean?
- [ ] Add Glean code generation to the build setup
- [ ] Add glean file linting to PRs

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,14 +27,18 @@
     "prepare-translation-status": "tsx scripts/prepare-translation-status.ts",
     "generate-pwa-icons": "tsx scripts/generate-pwa-icons.ts",
     "postinstall": "ignore-dependency-scripts \"stale-dep -u && simple-git-hooks && nuxi prepare && nr prepare-translation-status\"",
-    "release": "stale-dep && bumpp && tsx scripts/release.ts"
+    "release": "stale-dep && bumpp && tsx scripts/release.ts",
+    "build:glean": "glean translate metrics.yaml pings.yaml -f typescript -o glean",
```
Comment (bassrock) [inline package.json:31]: Glean generates code from a metrics.yml and pings.yml file

File: metrics.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Required to indicate this is a `metrics.yaml` file
+$schema: moz://mozilla.org/schemas/glean/metrics/2-0-0
+
+timeline:
```
Comment (bassrock) [inline metrics.yaml:4]: This file will eventually be defined in collaboration with @kirill-demtchouk 's team

File: modules/glean/runtime/glean-plugin.client.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import Glean from '@mozilla/glean/web'
+import { page } from '../../../telemetry/generated/pings'
+
+export default defineNuxtPlugin((nuxtApp) => {
+  nuxtApp.hook('page:start', () => {
+    // TODO: add in client id, logging on dev, upload off on dev, page path params to the ping/metric
+    Glean.initialize('mozilla-social.elk', true, {})
```
Comment (bassrock) [inline modules/glean/runtime/glean-plugin.client.ts:7]: Need to figure out the right nuxt hooks.

File: modules/glean/runtime/glean-plugin.client.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+import { page } from '../../../telemetry/generated/pings'
+
+// MDN Code for reference https://github.com/mdn/yari/blob/d11b0a8ba59d6c68e763eb8922ed7ce12fef2cae/client/src/telemetry/glean-context.tsx#L100
+export default defineNuxtPlugin((nuxtApp) => {
+  nuxtApp.hook('app:mounted', () => {
+    log.info('Glean: App Mounted, start initing glean')
+
+    // TODO: add in client id, logging on dev, upload off on dev, page path params to the ping/metric
+    Glean.initialize('mozilla-social.elk', true, {})
+    Glean.setLogPings(true)
```
Comment (Dexterp37) [inline modules/glean/runtime/glean-plugin.client.ts:12]: Note that this is only useful for debugging purposes

File: package.json [hunk 5]
```diff
@@ -27,14 +27,18 @@
...
+    "docs:glean": "glean translate telemetry/metrics.yaml pings.yaml -f markdown -o telemetry/generated/docs",
+    "lint:glean": "glean glinter telemetry/metrics.yaml telemetry/pings.yaml"
   },
   "dependencies": {
     "@emoji-mart/data": "^1.1.2",
     "@fnando/sparkline": "^0.3.10",
     "@iconify-emoji/twemoji": "^1.0.2",
     "@iconify/json": "^2.2.85",
     "@iconify/utils": "^2.1.7",
+    "@mozilla/glean": "^1.4.0",
```
Comment (Dexterp37) [inline package.json:41]: A more recent version of glean is out, it's strongly recommended to start with that one :)

File: telemetry/generated/element.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// AUTOGENERATED BY glean_parser v7.2.1. DO NOT EDIT. DO NOT COMMIT.
```
Comment (Dexterp37) [inline telemetry/generated/element.ts:5]: We recommend to not vendor the generated files, but rather generate them as part of the deployment process (for a variety of reasons, including playing nice with local linters and the fact that they are not necessarily stable across glean updates). Just a recommendation though: your call!

File: telemetry/pings.yaml [hunk 7]
```diff
@@ -0,0 +1,27 @@
+---
+# Schema
+$schema: moz://mozilla.org/schemas/glean/pings/2-0-0
+
+page:
```
Comment (Dexterp37) [inline telemetry/pings.yaml:5]: Your use-case can be much simpler! No need for pings.yaml, just need metrics.yaml with _events_. Other properties can be attached to your events as you see fit.

**Pull request closed without merging**